### DISCUSSION TITLE
link to wildcards inside wildcards?

### DISCUSSION THREAD

**Discussion by lgwjames** [lllyasviel/Fooocus#3493] [Q&A] [answered]
is there a way to have wildcards see inside folders?
as you can see in this photo, when i downloaded a wildcard of a site, all the camera types are in other folders.

be great if we could have a folder like clothes and add loads of clothes inside then use one txt file, if you get my meaning

![Screenshot 2024-08-11 095149](https://github.com/user-attachments/assets/93fa8092-f964-4c1b-96f4-be11602a6905)

**Comment by mashb1t** [comment]
That's currently not possible wildcards have to be located in the root folder.
I currently do not have time for further feature development, but it shouldn't be hard to extend the logic. Feel free to give it a shot!

**Comment by Turpo69** [comment, marked as answer]
It reads subfolder files just fine. If you just added them to wildcards folder then you need to restart Fooocus to use them. Or maybe Refresh All Files in the Models tab do the trick.

**Reply by mashb1t** [reply]
~~Maybe i've lost oversight in my own code, but where's the code to handle subfolders?~~

Nvm, it supports wildcards in subfolders, sorry.

**Comment by lgwjames** [comment]
thanks folks, will look later
